### ISSUE TITLE
Increase the refresh rate for the counters -\LogicalDisk(*)\Free Megabytes- and -\LogicalDisk(*)\% Free Space

### ISSUE THREAD

**Issue by WhiteJackk** [XhmikosR/perfmonbar#18]
This PERFMONBAR is a very nice program.
Would it be possible to increase the refresh rate for the counters --"\LogicalDisk(*)\Free Megabytes"-- and --"\LogicalDisk(*)\% Free Space"-- ?
At the moment they seem to update about ,maybe, every 3 minutes.
There seem to be very few programs that show free disk space as a number on the taskbar/system tray - one I have come across is (the perhaps obscure) -freespacemonitor-
 -- https://web.archive.org/web/20181117014923/https://solvalou.com/downloads/freespacemonitor.php -- .
Freespacemonitor updates about ,maybe, every 5 seconds.It shows,in the system tray,in one icon ,the free space as a number in GB and below that the free space as a percentage.Mouseover the systray icon and it shows in a tooltip the free space in MB -this can be done to easily verify that the figures are being regularly updated.
Another program is -traydiskfree-  -- https://sourceforge.net/projects/traydiskfree/  -- .I have not tried traydiskfree -it requires old visual basic runtimes and perhaps registering dll files and whatever else.
The obvious problem with showing a number in a single icon is space -the lack of.Therefore,the paucity of programs,no?
There are ,of course, programs that place icons in the systray in the form of bars or pie charts ,and so on, that will show the free disk space as a number when hovered over.
If the counters --"\LogicalDisk(*)\Free Megabytes"-- and --"\LogicalDisk(*)\% Free Space"-- are chosen, I think it would be a good idea to have PERFMONBAR update them every 5 seconds or so.
However,if this is not possible then ;"no problem", and if it is possible but you don't want to do it ,again ;"no problem".
I have a 256 GB ssd - I like to keep track of free space.
Is PERFMONBAR using a Windows default for polling the aforementioned counters ,while other programs actively poll them ??
Thanks for reading.
Oh , I should mention that in PERFMONBAR I have --"\LogicalDisk(*)\Free Megabytes"-- showing in GB with 3 decimals and --"\LogicalDisk(*)\% Free Space"-- showing 2 decimals.

**Comment by cmconti** [comment]
It's poorly documented, but it appears that the LogicalDisk Free Megabytes and Free Space counters are only updated every 5 minutes

- https://social.msdn.microsoft.com/Forums/en-US/2b022510-27f3-4c57-a55c-7f092f169564/refresh-rate-for-logicaldisk-free-megabytes-performancecounter?forum=netfxbcl

- https://www.ibm.com/support/pages/freembytes-not-matching-value-perfmon

My guess is those other utilities are not using perf counters to get the data, but calling some other API

**Comment by WhiteJackk** [comment]
CMCONTI,
thank you for providing the information and links -it is most appreciated.

So , the update rate is 5 minutes -good to know.

The IBM link -- [https://www.ibm.com/support/pages/freembytes-not-matching-value-perfmon](https://www.ibm.com/support/pages/freembytes-not-matching-value-perfmon) -- provided a solution -a method for "more frequent measurements......of the Logical Disk % Free Space and Free Megabytes metrics" (taken from Chapter 3: Measuring Server Performance of Microsoft Windows Server 2003 Performance Guide).I tried it (Windows 7 -two computers) - I ran regedit as administrator then as system then as trusted installer.On each occasion the required key was created without any problem but no matter what the update rate I entered it had no effect - the counter refreshed only every 5 minutes.So, it does not work or I did something wrong or I misunderstood something.It is what it is.
